transport/httptransport/http_transport.go

```diff
@@ -0,0 +1,241 @@
+package httptransport
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	"github.com/filecoin-project/boost/transport/types/transferstatus"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/libp2p/go-eventbus"
+	"github.com/libp2p/go-libp2p-core/event"
+
+	"github.com/filecoin-project/boost/transport/httptransport/pb"
+	"github.com/filecoin-project/boost/transport/types"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/filecoin-project/boost/transport"
+)
+
+var log = logging.Logger("http-transport")
+
+const (
+	readBufferSize = 4096
+)
+
+var _ transport.Transport = (*httpTransport)(nil)
+
+type httpTransport struct {
+	// TODO Construct and inject an http client here ?
+}
+
+func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealInfo *types.TransportDealInfo) (th transport.Handler, isComplete bool, err error) {
+
+	// de-serialize transport opaque token
+	tInfo := &pb.HttpReq{}
+	if err := proto.Unmarshal(transportInfo, tInfo); err != nil {
+		return nil, false, fmt.Errorf("failed to de-serialize transport info bytes: %w", err)
+	}
+	// check that the outputFile exists
+	fi, err := os.Stat(dealInfo.OutputFile)
```

Comment:
Would it make sense to just create the file if it doesn't exist here?

Reply:
I strongly disagree. We want the caller to control the entire lifecycle of the output file and don't want that to get fragmented between the Boost SM & the transport. If down the road, it turns out that we do need this, we can add a config option but let's avoid for now.

Reply:
Ok sounds good 👍